packages/amplify_auth_cognito/lib/method_channel_auth_cognito.dart

```diff
@@ -299,9 +299,7 @@ class AmplifyAuthCognitoMethodChannel extends AmplifyAuthCognito {
         nextStep: AuthNextSignUpStep(
             signUpStep: res["nextStep"]["signUpStep"],
             codeDeliveryDetails: res["nextStep"]["codeDeliveryDetails"],
-            additionalInfo: res["nextStep"]["additionalInfo"] is String
-                ? jsonDecode(res["nextStep"]["additionalInfo"])
-                : {}));
+            additionalInfo: res["nextStep"]["additionalInfo"] ?? {}));
```

Comment:
I think that we need to change the Android [FlutterSignInResult](https://github.com/aws-amplify/amplify-flutter/blob/6bc64a7b2bd4f22944b4147d6aa65a6d533663d4/packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignInResult.kt#L30) class to accomodate this change.

From this (line 30):
```kotlin
 "additionalInfo" to Gson().toJson(raw.nextStep.additionalInfo),
```
to this:
```kotlin
"additionalInfo" to (raw.nextStep.additionalInfo ?: emptyMap()),
```
Have you found otherwise?

Reply:
Let me take a look. If so, I'll update the PR.

Reply:
Confirmed that this did need to be updated for Android. Added your proposed fix and confirmed it works.